src/main/resources/public/index.js

```diff
@@ -0,0 +1,59 @@
+const StoreList = {
+    data() {
+        return {
+            stores: [],
+            currentStore: null,
+            currentPage: 1,
+            nameSearch: null,
+            newStoreName: null,
+        }
+    },
+    watch: {
+        nameSearch() {
+            this.getStores()
+            this.currentPage = 1
+            this.currentStore = null
+        },
+        currentPage() {
+            this.currentStore = null
+            this.getStores()
+        },
+        currentStore(val) {
+            if (val && val.name === null) val.name = '';
+            this.newStoreName = null
+        },
+    },
+    methods: {
+        nextPage() {
+            this.currentPage++
+        },
+        previousPage() {
+            this.currentPage--
+        },
+        getStores() {
+            axios
+                .get('/stores', {
+                    params: {
+                        page: this.currentPage,
+                        nameSearch: this.nameSearch,
+                    }
+                })
+                .then(response => this.stores = response.data)
+        },
+        confirmStoreName(storeId) {
+            axios
```

Comment:
what happens if this request fails?

Reply:
the app fails silently 😅
I should have shown a warning to the user.
depending on the error category, I could try a retry before that.